src/test/java/com/atlassian/bitbucket/jenkins/internal/client/ModernBitbucketBuildStatusClientImplTest.java

```diff
@@ -0,0 +1,111 @@
+package com.atlassian.bitbucket.jenkins.internal.client;
+
+import com.atlassian.bitbucket.jenkins.internal.model.BitbucketBuildStatus;
+import com.atlassian.bitbucket.jenkins.internal.model.BuildState;
+import com.atlassian.bitbucket.jenkins.internal.model.TestResults;
+import com.atlassian.bitbucket.jenkins.internal.provider.InstanceIdentityProvider;
+import com.atlassian.bitbucket.jenkins.internal.util.TestUtils;
+import jcifs.util.Base64;
+import okhttp3.HttpUrl;
+import org.apache.commons.lang3.StringUtils;
+import org.jenkinsci.main.modules.instance_identity.InstanceIdentity;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.ArgumentMatchers;
+import org.mockito.Captor;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import javax.annotation.Nullable;
+
+import java.security.KeyPair;
+import java.security.Signature;
+import java.security.interfaces.RSAPrivateKey;
+import java.security.interfaces.RSAPublicKey;
+import java.util.Map;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.*;
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+@RunWith(MockitoJUnitRunner.class)
+public class ModernBitbucketBuildStatusClientImplTest {
+
+    private static final String BASE_URL = "http://localhost:7990/bitbucket";
+    private static final String SHA1 = "5ab78046a50050b8aa3e4accf80950a60f716391";
+
+    @Captor
+    ArgumentCaptor<Map<String, String>> captor;
+    ModernBitbucketBuildStatusClientImpl client;
+    @Mock
+    BitbucketRequestExecutor executor;
+    @Mock
+    InstanceIdentityProvider identityProvider;
+    @Mock
+    InstanceIdentity instanceIdentity;
+    KeyPair keyPair;
+
+    @Before
+    public void init() {
+        keyPair = TestUtils.createTestKeyPair();
```

Comment:
I imagine it would be desirable to do it once for the class and just re-use the keys for every test?

Reply:
Good point, cheers